todo_app/app.py

```diff
@@ -2,44 +2,95 @@
 from operator import itemgetter
 
 from todo_app.flask_config import Config
-from todo_app.data.session_items import get_items, get_item, add_item, save_item, remove_item
+from todo_app.data.trello_items import get_trello_credentials, get_trello_board_id, get_trello_cards, get_trello_list_id, get_trello_lists_on_board, create_trello_card, TrelloCard, move_trello_card, archive_trello_card
 
 app = Flask(__name__)
 app.config.from_object(Config)
 
 
 @app.route('/')
 def index():
-    items = get_items()
+
+    trello_list_id = {}
+    trello_list_id['To Do'] = get_trello_list_id('To Do')
+    trello_list_id['In Progress'] = get_trello_list_id('In Progress')
+    trello_list_id['Completed'] =get_trello_list_id('Completed')
+
+    all_lists = get_trello_lists_on_board()
+
+    items = get_trello_cards()
     if request.values.get('sort') == '1':
-        items = sorted(items, key=itemgetter('status'))
+        items.sort(key=lambda x: x.idList)
     elif request.values.get('sort') == '2':
-        items = sorted(items, key=itemgetter('status'), reverse=True)
-    return render_template("index.html",items=items)
+        items.sort(key=lambda x: x.idList, reverse=True)
+    return render_template("index.html",items=items,lists=all_lists)#
 
-@app.route('/', methods=['POST'])
-def add_list():
-    new_item_title = request.form.get('title')
-    add_item(new_item_title)
-    return redirect(('/'))
+@app.route('/new_item', methods=['POST'])
+def new_item():
+    new_item_title = request.form.get('new_item_title')
+    trello_default_list = get_trello_list_id('To Do')
+    
+    new_card = TrelloCard(0, new_item_title, trello_default_list)
+    create_trello_card(new_card)
+    return redirect(request.headers.get('Referer'))
 
-@app.route('/remove_item', methods=['POST'])
+@app.route('/archive_item', methods=['POST'])
```

Comment:
There's actually a delete card endpoint on the Trello API. Any reason you're archiving instead?

Even if you do archive the card, there's no real need to call it archiving in the route. As far as your app is concerned it's deleted since your app doesn't interact with archived cards.

Reply:
It was more from being able to reference old tasks. I get what you are saying though... I have changed this to delete rather than archive.